### PR TITLE
oc adm: find and sanitize `Bug` and `release` in PR titles

### PR DESCRIPTION
**Pull request by eparis**
This will strip all [release-N.N] tags.
```
* [release-4.1] this is a title
     becomes
* this is a title
```
It will find `Bug NNNNNNNN:` anywhere (the `:` makes this pretty safe) and make it a link
It will create bright red things about a revert:
```
* <span style="color:red">REVERT</span> [Bug 1717640](https://bugzilla.redhat.com/show_bug.cgi?id=1717640): Add related objects to Status [#39](https://github.com/openshift/cluster-storage-operator/pull/39) <span style="color:red">REVERT</span>
* [Bug 1717640](https://bugzilla.redhat.com/show_bug.cgi?id=1717640): Add related objects to Status [#38](https://github.com/openshift/cluster-storage-operator/pull/38)
```
It will escape any html in a PR title
It will escape `` in PR titles

### CODE REVIEW

**Comment by eparis** [comment]
@smarterclayton I think you wrote all of this.

**Comment by smarterclayton** [comment]
The original intent here was to force good PR titles for the changelog, so a human could actually read it.  Allowing bugs everywhere doesn’t really fit that.

The bug discussed stripping all [.] prefixes, if we could prove that we wouldn’t want those in the changelog.

**Comment by smarterclayton** [comment]
I don’t really want Wild West changelogs.  It’s not hard to use a little discipline in PR titles, since we show them to customers and users by default.

File: staging/src/github.com/openshift/oc/pkg/cli/admin/release/git.go [hunk 1]
```diff
@@ -205,16 +198,21 @@ func mergeLogForRepo(g *git, repo string, from, to string) ([]MergeCommit, error
 		}
 
 		msg := records[3]
-		if m := reBug.FindStringSubmatch(msg); m != nil {
-			mergeCommit.Subject = msg[len(m[0]):]
-			mergeCommit.Bug, err = strconv.Atoi(m[1])
-			if err != nil {
+
+		// If it contains `[release-X.Y]` anywhere, just throw that away
```
Comment (smarterclayton) [inline staging/src/github.com/openshift/oc/pkg/cli/admin/release/git.go]: This doesn’t work for repos that have openshift-* branches (etcd, prometheus)

Comment (smarterclayton) [reply staging/src/github.com/openshift/oc/pkg/cli/admin/release/git.go]: Stripping the leading [xxx] segments until we hit something that isn’t guarded that way might be a compromise I could live with, although it’s a bit precious.

Comment (smarterclayton) [reply staging/src/github.com/openshift/oc/pkg/cli/admin/release/git.go]: It also wouldn’t work for hot-fix branches or those that were created for an embargo

**Comment by smarterclayton** [comment]
As discussed offline if you can add unit tests (and potentially split the transformation function out so it can be better tested) I can survive relaxing this to allow people to provide any old hacky title.

**Pull request closed without merging**